storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -116,11 +116,7 @@ func (s *fileSystemSource) enqueueReaders(
 	readersCh chan<- shardReaders,
 ) {
 	for shard, tr := range shardsTimeRanges {
-		var files []string
-		fs.ForEachInfoFile(s.filePathPrefix, namespace, shard, s.readerBufferSize, func(fname string, _ []byte) {
-			files = append(files, fname)
-		})
-
+		files := fs.ReadInfoFiles(s.filePathPrefix, namespace, shard, s.readerBufferSize)
```

Comment:
I wonder if it's possible instead to dictate what kind of validation/data reading we perform by passing in an enum flag etc so that we don't perform additional unnecessary operation if we don' thave